src/components/scenes/SwapCreateScene.tsx

```diff
@@ -126,8 +126,12 @@ export const SwapCreateScene = (props: Props) => {
     else if (changed === 'amount' && ['InsufficientFundsError', 'SwapAboveLimitError', 'SwapBelowLimitError'].includes(error.name)) {
       clearError = true
     }
-    // Selected asset related errors
-    else if (changed === 'asset' && ['SwapPermissionError', 'SwapCurrencyError'].includes(error.name)) {
+    // Selected asset related errors (arbitrarily includes all amount-related
+    // errors as well)
+    else if (
+      changed === 'asset' &&
+      ['SwapPermissionError', 'SwapCurrencyError', 'InsufficientFundsError', 'SwapAboveLimitError', 'SwapBelowLimitError'].includes(error.name)
+    ) {
```

Comment:
I somehow missed this on the previous review, but in the lines about, we should be using the core-provided cleaners `asMaybeInsufficientFundsError`, `asMaybeSwapAboveLimitError`, etc..., instead of directly checking `error.name`.

We could do it this way:

```ts
const isSwapPermission = asMaybeSwapPermissionError(error) != null
const isInsufficentFunds = asMaybeInsufficientFundsError(error) != null
const isSwapCurrency = asMaybeSwapCurrencyError(error) != null
...etc...

if (changed === 'asset' &&
  (isSwapPermission || isSwapCurrency || isInsufficentFunds || isSwapAboveLimit || isSwapBelowLimit)) {
    clearError = true
}
```

Not sure if this change is super-duper-necessary, but it would be a bit "more correct".